src/Plugin.Maui.Audio/AudioPlayer/AudioPlayer.macios.cs

```diff
@@ -81,6 +82,31 @@ internal AudioPlayer(string fileName)
 		PreparePlayer();
 	}
 
+	static void InitAudioSession()
+	{
+		var audioSession = AVAudioSession.SharedInstance();
+
+		// AVAudioSessionCategoryOptions.DefaultToSpeaker ensures no other audio is playing
+		var error = audioSession.SetCategory(AVAudioSessionCategory.Playback, AVAudioSessionCategoryOptions.DefaultToSpeaker);
+		if (error is not null)
+		{
+			Console.WriteLine(error.ToString());
+			//throw new FailedToPlayAudioException(error.ToString());
+		}
+
+		error = audioSession.SetActive(true, AVAudioSessionSetActiveOptions.NotifyOthersOnDeactivation);
```

Comment:
What is the implication of this? Will it stop other music from playing? If so then I would like to consider whether we can make this configurable. I know I've fought with that behavior in the past when wanting to play sound effects in a game and still allow users to play their music.

Reply:
Oh yes, it is. Could be passed as an enum option instead, maybe.
There are a few options:
Stop other audio, reduce the noise of other audio, or mix entirely with other audio. I don’t know about android or windows though.

Reply:
Perhaps we could/should solve this in a separate PR just to make it easier to track the changes. I like the idea of an options class and we could create an `AudioRecorderOptions.macios.cs` file that contains this. That would enable developers to specify those values on iOS. What do you think?

Reply:
Do you mean to add another AudioPlayerOptions class?
Because the options for play should probably not be mixed with the recorder options, right?
And yeah I guess it would be possible to split things up and give different options for each platform. On the other hand I think in an ideal world there would be a way to map the most common use cases (or all of them) to the different platforms?

One thing I am already uncertain about are the AirPlay and Bluetooth options. Because it seems like you can only set one Category on iOS, but then what happens about the other audio options (does it mix/ stop/ duck other audio) when you choose allow AirPlay.

Options could maybe look similar to this:

Enum OtherAudio Mix/Duck/Stop/TryContinueOtherAudio/AllowBluetooth/AllowStreaming(AirPlay and similar things)

TryContinueOtherAudio would mean other players would be notified on session end, but it depends whether the other audio player listens to that.

Also considerable would be

bool EndSessionOnFinish

but maybe it's not necessary with well defined enum options.

Reply:
Yes I think we would want separate options for AudioPlayer and AudioRecorder. There is a bug in the player code where it just relies on the defaults which gets messed if you start recording. I vote we go with sensible defaults for each operation (play, record) and then allow the ability to override it for Apple platforms atm. I would like to do this in a separate PR though. Are you OK if I move some of that into a new PR? Do you like the idea?

Reply:
Yes I agree that's probably best. Should've probably separated the big fixes and the options, sorry!
Feel free to dismiss or modify this PR and just take whatever you want!

Reply:
Yes I agree that's probably best. Should've probably separated the bug fixes and the options, sorry!
Feel free to dismiss or modify this PR and just take whatever you want!